src/OpenTelemetry.Contrib.EventCounterListener/README.md

```diff
@@ -0,0 +1,58 @@
+# MySqlData Instrumentation for OpenTelemetry
+
+This is an
+[Instrumentation Library](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/glossary.md#instrumentation-library),
+which republishes EventCounters using Metrics Api. 
+
+## Steps to enable OpenTelemetry.Contrib.Instrumentation.EventCounterListener
+
+### Step 1: Install Package
+
+Add a reference to the
+[`OpenTelemetry.Contrib.EventCounterListener`](https://www.nuget.org/packages/OpenTelemetry.Contrib.EventCounterListener)
+package. Also, add any other instrumentations & exporters you will need.
+
+```shell
+dotnet add package OpenTelemetry.Contrib.EventCounterListener
+```
+
+### Step 2: Enable EventCounterListener Instrumentation at application startup
+
+EventCounterListener instrumentation must be enabled at application startup.
+
+The following example demonstrates adding EventCounter events to a
+console application. This example also sets up the OpenTelemetry Console
+exporter, which requires adding the package
+[`OpenTelemetry.Exporter.Console`](https://www.nuget.org/packages/OpenTelemetry.Exporter.Console)
+to the application.
+
+```csharp
+using OpenTelemetry;
+using OpenTelemetry.Metrics;
+using OpenTelemetry.Contrib.Instrumentation.EventCounterListener;
+
+namespace DotnetMetrics
+{
+    public class Program
+    {
+
+        public static void Main(string[] args)
+        {
+            using var meterprovider = Sdk.CreateMeterProviderBuilder()
+                    .AddEventCounterListener(options =>
+                    {
+                        options.Sources = null;
+                    })
+                    .AddConsoleExporter()
+                    .Build();
+
+            System.Threading.Thread.Sleep(15000); // Give it some time to record metrics
+
+        }
+    }
+}
+```
+
```

Comment:
the usage is not clear... could you describe with an example?

Reply:
If you are looking for how configuration is done, I havent coded that up yet, just making sure what I have so far is in the ballpark. Using the extension method provided (`AddEventCounterListener`) adds the EventListener that then republishes counters via metricsAPI. For example running the program listing in the `Readme.md` above produces the following output:

- I assumed all counters will be instrumented as guages 
- Since this requires aggregation in the `collect` interval, the appropriate one is last value

`Export cpu-usage, Unit: CPU Usage, Meter: OpenTelemetry.Contrib.EventCounterListener/0.0.0.0                                                                                                                                                 (2022-01-20T03:48:23.3248145Z, 2022-01-20T03:48:37.2994387Z] DoubleGauge                                                                                                                                                                     Value: 0                                                                                                                                                                                                                                                                                                                                                                                                                                                                                  Export working-set, Unit: Working Set, Meter: OpenTelemetry.Contrib.EventCounterListener/0.0.0.0                                                                                                                                             (2022-01-20T03:48:23.3275819Z, 2022-01-20T03:48:37.2994398Z] DoubleGauge                                                                                                                                                                     Value: 31                                                                                                                                                                                                                                                                                                                                                                                                                                                                                 Export gc-heap-size, Unit: GC Heap Size, Meter: OpenTelemetry.Contrib.EventCounterListener/0.0.0.0                                                                                                                                           (2022-01-20T03:48:23.3276834Z, 2022-01-20T03:48:37.2994400Z] DoubleGauge                                                                                                                                                                     Value: 5                                                                                                                                                                                                                                                                                                                                                                                                                                                                                  Export gen-0-gc-count, Unit: Gen 0 GC Count, Meter: OpenTelemetry.Contrib.EventCounterListener/0.0.0.0                                                                                                                                       (2022-01-20T03:48:23.3280650Z, 2022-01-20T03:48:37.2994402Z] DoubleGauge                                                                                                                                                                     Value: 0   `

Reply:
without seeing how a user would configure, its hard to comment on it.
Some qns:
What are the inputs provided by the user? Do they pick each and every eventcounter?
Given there are different types of EventCounters, do we let user let us know if they want it treated as counter/gauge/histogram?

(Ofcourse we don't want to solve it in 1st iteration, but want to get a feel of how would the final version look like )

Reply:
> What are the inputs provided by the user? Do they pick each and every eventcounter?

I was thinking they can filter by EventSource, and then get more granular at the particular counter. 
>Given there are different types of EventCounters, do we let user let us know if they want it treated as counter/gauge/histogram?

I have made a list of [well known eventcounters](https://docs.microsoft.com/en-us/dotnet/core/diagnostics/available-counters) and ran [Reiley's "algorithm" for picking an instrument](https://github.com/open-telemetry/opentelemetry-specification/pull/1617#issue-858179203). In each case they made no sense to add them, since the eventCounters themselves are preaggregated, it makes no sense to further aggregate them - let me know if I am wrong about this or a different guide to pick instruments. Dependending on the answer to this,  we could also make this the default option while giving users customizability to change this. 

If you need to get on a zoom call for 15 mins, I am available today or next week is fine as well.

Reply:
I did take a crack at what usage might look like if it needs to be able to support multiple instrument types. However if we can ascertain that we dont need to use other instrument types because of the nature of eventcounters, then everything will be very simple.  
```csharp
   this.meterProvider = Sdk.CreateMeterProviderBuilder()
                 .AddEventCounterListener(options =>
                 {
                    options.Sources = new[]
                    {
                        new EventSourceOption
                        {
                            EventSourceName = "System.Runtime",
                            RefreshInterval = "1",
                            EventCounters = new[] { new EventCounter{ Name = "cpu-usage", Type = "ObservableGuage" } },
                        },
                    };
                 })
                .AddReader(metricReader)
                .Build();
```

Reply:
I did realize something I had misunderstood about different eventcounter types. I'll rework this.